system-addon/bootstrap.js

```diff
@@ -156,4 +157,11 @@ this.shutdown = function shutdown(data, reason) {
   modulesToUnload.forEach(Cu.unload);
 };
 
-this.uninstall = function uninstall(data, reason) {};
+this.uninstall = function uninstall(data, reason) {
+  // Uninstall is also called before an upgrade, so we
+  // need to make sure this is an actual uninstall
+  if (activityStream && reason === REASON_ADDON_UNINSTALL) {
+    activityStream.uninstall(reason);
```

Comment:
The `reset` stuff removes *user* set prefs, which we don't want to do on an upgrade/downgrade – default prefs get cleared on their own on a browser shut down, since they aren't persisted anywhere